### ISSUE TITLE
Date arrays and server-side binding

### ISSUE THREAD

**Issue by ewjoachim** [ClickHouse/clickhouse-connect#188]
### Describe the bug
Same as #159 but with arrays of dates.

### Steps to reproduce
```python
import datetime
client.query('SELECT {l:Array(Date)}', parameters={"l": [datetime.date.today()]}).result_rows
```
```
DatabaseError: :HTTPDriver for https://rw.chcl5l17008fjmn0m8s9.at.double.cloud:8443/ returned response code 400)
 Code: 27. DB::ParsingException: Cannot parse input: expected '\'' before: '2023-05-05]'. (CANNOT_PARSE_INPUT_ASSERTION_FAILED) (version 22.8.17.17 (official build))
```
### Expected behaviour

returns
```python
[([datetime.date(2023, 5, 5)],)]
```

### Code example
```python
import clickhouse_connect
client = clickhouse_connect.get_client(
    dsn=connection_string
)
client.query('SELECT {l:Array(Date)}', parameters={"l": [datetime.date.today()]}).result_rows
```

#### Environment
* clickhouse-connect version: 0.5.23
* Python version: 3.7
* Operating system: Ubuntu

#### ClickHouse server
* ClickHouse Server version:22.8.15.23

**Comment by genzgd** [comment]
Thanks for the detailed report!  This should be fixed in 0.5.24

**Comment by ewjoachim** [comment]
Thanks !
